### PR TITLE
Base arch structure

### PR DESCRIPTION
**Pull request by WilliamHarvey97**
- Create a project example using this arch https://github.com/PolyCortex/polydodo/wiki/Mobile-Architecture
- Add some important dependencies

### CODE REVIEW

File: mobile/lib/test/widget_test.dart [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import 'package:flutter_test/flutter_test.dart';
```
Comment (abelfodil) [inline mobile/lib/test/widget_test.dart:1]: move the `test` folder to `mobile/`

File: mobile/lib/src/app.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/material.dart';
```
Comment (abelfodil) [inline mobile/lib/src/app.dart:1]: Move `lib/src` content to `lib/`

File: mobile/lib/test/widget_test.dart (hunk 1 above)
Comment (WilliamHarvey97) [reply mobile/lib/test/widget_test.dart:1]: oups effectivement

File: mobile/lib/src/app.dart (hunk 2 above)
Comment (WilliamHarvey97) [reply mobile/lib/src/app.dart:1]: ### lib directory

    As you might expect, the library code lives under the lib directory and is public to other packages. You can create any hierarchy under lib, as needed. By convention, implementation code is placed under lib/src. Code under lib/src is considered private; other packages should never need to import src/.... To make APIs under lib/src public, you can export lib/src files from a file that’s directly under lib.

https://dart.dev/guides/libraries/create-library-packages

Comment (abelfodil) [reply mobile/lib/src/app.dart:1]: TIL

**Comment by abelfodil** [comment]
Tested on android emulator and doesn't crash.

LGTM

**Review by abelfodil — APPROVED** [review]

**Comment by MouradLachhab** [comment]
I was reading through the bloc doc, and I was wondering if we were going to use the bloc tests ? If so, we would need to add these to pubspec.yaml.
test: ^1.3.0
bloc_test: ^7.0.0

**Comment by WilliamHarvey97** [comment]
> I was reading through the bloc doc, and I was wondering if we were going to use the bloc tests ? If so, we would need to add these to pubspec.yaml.
> test: ^1.3.0
> bloc_test: ^7.0.0

Will we really test our blocs/cubits? I tought a little bit about it and I told myself that the most important thing to test is our domain layer since it is there that the app business logic will be concentrated

**Pull request merged**